activiti-cloud-query-service/activiti-cloud-services-query/activiti-cloud-services-query-events/src/test/java/org/activiti/cloud/services/query/events/handlers/IntegrationErrorReceivedEventHandlerTest.java

```diff
@@ -0,0 +1,142 @@
+import java.util.Collections;
+import java.util.UUID;
+import org.activiti.cloud.api.process.model.CloudBPMNActivity.BPMNActivityStatus;
+import org.activiti.cloud.api.process.model.CloudIntegrationContext.IntegrationContextStatus;
+import org.activiti.cloud.api.process.model.events.CloudIntegrationErrorReceivedEvent;
+import org.activiti.cloud.api.process.model.impl.events.CloudIntegrationErrorReceivedEventImpl;
+import org.activiti.cloud.services.query.model.IntegrationContextEntity;
+import org.activiti.cloud.services.query.model.ServiceTaskEntity;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class IntegrationErrorReceivedEventHandlerTest extends IntegrationEventsHelper {
+
+    @InjectMocks
+    private IntegrationErrorReceivedEventHandler errorHandler;
+
+    @Mock
+    private EntityManager entityManager;
+
+    @Test
+    public void handleShouldUpdateStatusesWhenIntegrationErrorReceived() {
+        // given
+        String id = UUID.randomUUID().toString();
+        IntegrationContextEntity existingIntegrationContextEntity = createIntegrationContextEntity(id);
+
+        existingIntegrationContextEntity.setServiceTask(createServiceTaskEntity(id));
+
+        when(entityManager.find(IntegrationContextEntity.class, id)).thenReturn(existingIntegrationContextEntity);
+
+        // when
+        CloudIntegrationErrorReceivedEvent resultEvent = buildIntegrationErrorReceivedEvent(id);
+        errorHandler.handle(resultEvent);
+
+        // then
+        ArgumentCaptor<IntegrationContextEntity> integrationContextCaptor = ArgumentCaptor.forClass(
+            IntegrationContextEntity.class
+        );
+        ArgumentCaptor<ServiceTaskEntity> serviceTaskCaptor = ArgumentCaptor.forClass(ServiceTaskEntity.class);
+
+        verify(entityManager, times(2)).persist(any());
+        verify(entityManager).persist(integrationContextCaptor.capture());
+        verify(entityManager).persist(serviceTaskCaptor.capture());
+
+        IntegrationContextEntity updatedIntegrationContext = integrationContextCaptor.getValue();
+        ServiceTaskEntity updatedServiceTask = serviceTaskCaptor.getValue();
+
+        assertThat(updatedIntegrationContext.getStatus())
+            .isEqualTo(IntegrationContextStatus.INTEGRATION_ERROR_RECEIVED);
+        assertThat(updatedIntegrationContext.getErrorDate()).isNotNull();
+
+        assertThat(updatedServiceTask.getStatus()).isEqualTo(BPMNActivityStatus.ERROR);
+        assertThat(updatedServiceTask.getCompletedDate()).isNull();
+    }
+
+    @Test
+    public void handleShouldUpdateStatusesForLegacyIdsWhenIntegrationResultReceived() {
+        // given
+        String new_uuid_id = UUID.randomUUID().toString();
+        String legacy_composite_key_id = getLegacyId();
```

Comment:
```suggestion
        String newUuid = UUID.randomUUID().toString();
        String legacyCompositeKeyId = getLegacyId();
```

Camel-case instead of snake case (remember to change in other places as well).